### ISSUE TITLE
Expiring anonymous genesets?

### ISSUE THREAD

**Issue by ravila4** [biothings/mygeneset.info#67]
Anonymous genesets are intended to allow quick sharing or user testing of the application, but we want to encourage users to associate an account for long-term storage. Therefore, we could set anonymous genesets to have a default expiration date of, say, 1 month (but we could allow customization for shorter or slightly longer expiries).

This could also help us discourage abuse and creation of spam or low quality genesets.

**Comment by ravila4** [comment]
The simplest implementation option might be to use Elasticsearch's index lifecycle management to delete indices when they reach a certain size. We would need to create a separate index for anonymous genesets.

https://www.elastic.co/guide/en/elasticsearch/reference/8.5/index-lifecycle-management.html

**Comment by vincerubinetti** [comment]
Let me know if you do this, because we'll want to have a big warning about this, and also list it as a "pro" for signing up.

I'd recommend being very lenient though. Maybe 3-6 months. Maybe someone wants to have a geneset ready for a conference they are attending later in the year, and creates one anonymously because they are too busy to create an account.

**Comment by ravila4** [comment]
Yes, probably not soon, I think this would be considered an enhancement. If we get a lot of anonymous genesets being created, we can reconsider.
